[bitnami/redis-cluster] Add possibility to set `hostNetwork: true` for redis pods

cjose3 opened the pull request:
<!--
 Before you open the request please review the following guidelines and tips to help it be more easily integrated:

 - Describe the scope of your change - i.e. what the change does.
 - Describe any known limitations with your change.
 - Please run any tests or examples that can exercise your modified code.

 Thank you for contributing! We will try to test and integrate the change as soon as we can, but be aware we have many GitHub repositories to manage and can't immediately respond to every request. There is no need to bump or check in on a pull request (it will clutter the discussion of the request).

 Also don't be worried if the request is closed or not integrated sometimes the priorities of Bitnami might not match the priorities of the pull request. Don't fret, the open source community thrives on forks and GitHub makes it easy to keep your changes in a forked repo.
 -->

**Description of the change**
Allow setting `hostNetwork: true` for pods of the redis statefulset. 

<!-- Describe the scope of your change - i.e. what the change does. -->

**Checklist**
<!-- [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

marcosbc reviewed (changes requested):
Thanks for the contribution! The technical changes look good to me, although I would request for you to also update the `values.yaml` file to remove the duplicated `@section Redis&trade; Cluster Common parameters` section.

Note that we autogenerate the README table with [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm), and the changes to the README could be reverted automatically.

cjose3 commented:
Hey @marcosbc, thanks for the review. 

I've removed the duplicated section from `values.ymal` and also reverted changes on the `README.md` file, so it can get autogenerate with [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm).

marcosbc reviewed (approved):
LGTM! Thanks!
